specification/deviceprovisioningservices/resource-manager/Microsoft.Devices/stable/2022-12-12/iotdps.json

```diff
@@ -0,0 +1,2571 @@
+      "type": "object",
+      "properties": {
+        "status": {
+          "description": "The status of a private endpoint connection",
+          "enum": [
+            "Pending",
+            "Approved",
+            "Rejected",
+            "Disconnected"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "PrivateLinkServiceConnectionStatus",
+            "modelAsString": true
+          }
+        },
+        "description": {
+          "type": "string",
+          "description": "The description for the current state of a private endpoint connection"
+        },
+        "actionsRequired": {
+          "type": "string",
+          "description": "Actions required for a private endpoint connection"
+        }
+      },
+      "required": [
+        "status",
+        "description"
+      ]
+    },
+    "SharedAccessSignatureAuthorizationRule[AccessRightsDescription]": {
+      "type": "object",
+      "description": "Description of the shared access key.",
+      "properties": {
+        "keyName": {
+          "type": "string",
+          "description": "Name of the key."
+        },
+        "primaryKey": {
+          "type": "string",
+          "description": "Primary SAS key value."
+        },
+        "secondaryKey": {
+          "type": "string",
+          "description": "Secondary SAS key value."
+        },
+        "rights": {
+          "description": "Rights that this key has.",
+          "enum": [
+            "ServiceConfig",
+            "EnrollmentRead",
+            "EnrollmentWrite",
+            "DeviceConnect",
+            "RegistrationStatusRead",
+            "RegistrationStatusWrite"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "AccessRightsDescription",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "keyName",
+        "rights"
+      ]
+    },
+    "Resource": {
+      "description": "The common properties of an Azure resource.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource identifier."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource name.",
+          "pattern": "^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{2,49}[a-zA-Z0-9]$"
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "location": {
+          "type": "string",
+          "description": "The resource location."
+        },
+        "resourcegroup": {
+          "type": "string",
+          "description": "The resource group of the resource."
+        },
+        "subscriptionid": {
+          "type": "string",
+          "description": "The subscription id of the resource."
+        },
```

Comment:
Why are these required? These are always there in Operation Resource URI, not able to derive from there?

Reply:
The changes made here beside the addition of the "portalOperationsHostName" were all motivated by S360 alerts raised against our service. The alerts wanted our traffic to match the swagger backing it.

Reply:
We should remove them, and stop returning them, in this new API version, since they're not actually allowed according to the RPC.

Reply:
Please also refactor by not declaring resource, and inheriting "allOf": TrackedResource or ProxyResource instead from common types.

Reply:
Resource definitions - use TrackedResource or ProxyResource in commontypes
Instead of defining your own resource class with required top level properties, 'id', 'type', 'name', and 'location' and 'tags' (for TrackedResource) its recommended to inherit from ProxyResource or TrackedResource in common-types v2 or v3.json. (The v2 definition does NOT include systemData, and is otherwise similar to the v3 one).
[https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56](https://github.com/Azure/azure-rest-api-specs/blob/be6cd9ccfcb6ba08c1c206627026eabfbff31fc1/specification/common-types/resource-management/v3/types.json#L56)